app/components/export-tool.js

```diff
@@ -0,0 +1,65 @@
+import Ember from 'ember';
+
+export default Ember.Component.extend({
+    attributeBindings: ['data', 'mappingFunction'],
+    dataFormat: 'JSON',
+    dataFormats: [ 'JSON', 
+        'CSV', 
+    ],
+    defaultMappingFunction: function (model) {
+        return model._data;
+    },
+    processedData: Ember.computed( 'data', 'mappingFunction', 'dataFormat', {
+        get() {
+            var dataArray = this.get('data');
+            var dataFormat = this.get('dataFormat');
+            var mappingFunction = this.get('mappingFunction');
+            if (Ember.isPresent(dataArray) && Ember.isPresent(mappingFunction)) {
+                var mapped = dataArray.map(mappingFunction);
+                return this.convertToFormat(mapped, dataFormat); 
+            } else if (Ember.isPresent(dataArray)) {
+                var mapped = dataArray.map(this.get('defaultMappingFunction'));
+                return this.convertToFormat(mapped, dataFormat); 
+            } else {
+                return null;
+            }
+        }, 
+        set(_, value) {
+            this.set('processedData', value);
```

Comment:
See http://emberjs.com/api/classes/Ember.ComputedProperty.html.

I had a bug earlier this morning where not returning value caused the computed to appear `undefined`